plugin/evm/vm.go

```diff
@@ -335,36 +356,63 @@ func (vm *VM) Initialize(
 	vm.genesisHash = vm.chain.GetGenesisBlock().Hash()
 	log.Info(fmt.Sprintf("lastAccepted = %s", lastAccepted.Hash().Hex()))
 
-	vm.State = chain.NewState(&chain.Config{
-		DecidedCacheSize:    decidedCacheSize,
-		MissingCacheSize:    missingCacheSize,
-		UnverifiedCacheSize: unverifiedCacheSize,
-		LastAcceptedBlock: &Block{
-			id:       ids.ID(lastAccepted.Hash()),
-			ethBlock: lastAccepted,
-			vm:       vm,
-			status:   choices.Accepted,
-		},
-		GetBlockIDAtHeight: vm.GetBlockIDAtHeight,
-		GetBlock:           vm.getBlock,
-		UnmarshalBlock:     vm.parseBlock,
-		BuildBlock:         vm.buildBlock,
-	})
+	vm.multiGatherer = avalanchegoMetrics.NewMultiGatherer()
+
+	// Initialize [vm.State]
+	if err := vm.initChainState(&Block{
+		id:       ids.ID(lastAccepted.Hash()),
+		ethBlock: lastAccepted,
+		vm:       vm,
+		status:   choices.Accepted,
+	}, metrics.Enabled); err != nil {
+		return err
+	}
 
 	vm.builder.awaitSubmittedTxs()
 	go vm.ctx.Log.RecoverAndPanic(vm.startContinuousProfiler)
 
 	// Only provide metrics if they are being populated.
 	if metrics.Enabled {
-		gatherer := prometheus.Gatherer(metrics.DefaultRegistry)
-		if err := ctx.Metrics.Register(gatherer); err != nil {
+		gatherer := subnetEVMPrometheus.Gatherer(metrics.DefaultRegistry)
+		if err := vm.multiGatherer.Register(ethMetricsPrefix, gatherer); err != nil {
 			return err
 		}
 	}
+	// Register [multiGatherer] after registerers have been registered to it
+	if err := ctx.Metrics.Register(vm.multiGatherer); err != nil {
+		return err
+	}
 
 	return nil
 }
 
+func (vm *VM) initChainState(lastAcceptedBlock *Block, metricsEnabled bool) error {
```

Comment:
note: we added these changes since we were no longer getting the equivalent metrics from the rpcchainvm. There's no harm in adding it in here and they may be slightly different since the difference will measure the rpcchainvm overhead.

Reply:
If not needed, let's remove. It is confusing.

Reply:
removed.